app/src/main/java/com/framgia/wsm/screen/requestleave/RequestLeavePresenter.java

```diff
@@ -73,83 +73,67 @@ public boolean validateDataInput(Request request) {
         switch (request.getLeaveType().getName()) {
             case RequestLeaveViewModel.LeaveType.IN_LATE_A:
             case RequestLeaveViewModel.LeaveType.IN_LATE_M:
-                if (validateCheckinTime(request.getCheckinTime())
-                        & validateCompensationFrom(request.getCompensation().getFromTime())
-                        & validateCompensationTo(request.getCompensation().getToTime())
-                        & validateReason(request.getReason())
-                        & validateProjectName(request.getProject())) {
+                if (validateCheckinTime(request.getCheckinTime()) & validateCompensationFrom(
+                        request.getCompensation().getFromTime()) & validateCompensationTo(
+                        request.getCompensation().getToTime()) & validateReason(
+                        request.getReason())) {
                     return true;
                 }
                 break;
             case RequestLeaveViewModel.LeaveType.LEAVE_EARLY_A:
             case RequestLeaveViewModel.LeaveType.LEAVE_EARLY_M:
-                if (validateCheckoutTime(request.getCheckoutTime())
-                        & validateCompensationFrom(request.getCompensation().getFromTime())
-                        & validateCompensationTo(request.getCompensation().getToTime())
-                        & validateReason(request.getReason())
-                        & validateProjectName(request.getProject())) {
+                if (validateCheckoutTime(request.getCheckoutTime()) & validateCompensationFrom(
+                        request.getCompensation().getFromTime()) & validateCompensationTo(
+                        request.getCompensation().getToTime()) & validateReason(
```

Comment:
ERROR: [STYLE][NS_DANGEROUS_NON_SHORT_CIRCUIT] Potentially dangerous use of non-short-circuit logic in com.framgia.wsm.screen.requestleave.RequestLeavePresenter.validateDataInput(Request)
http://findbugs.sourceforge.net/bugDescriptions.html#NS_DANGEROUS_NON_SHORT_CIRCUIT